test/node/user-presence.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const assert = require('assert');
+const Circuit = require('../../circuit-node');
+const config = require('./config.json');
+const helper = require('./helper');
+Circuit.logger.setLevel(Circuit.Enums.LogLevel.Error);
+
+let client;
+let user;
+let client2;
+let user2;
+let presence;
+describe('User Presence', () => {
+    before(async () => {
+        client = new Circuit.Client(config.bot1);
+        user = await client.logon();
+        client2 = new Circuit.Client(config.bot2);
+        user2 = await client2.logon();
+    });
+
+    after(async () => {
+        await client.logout();
+        await client2.logout();
+    });
+
+    it('should subscribe to  users presence', async () => {
+        const res = await client.subscribePresence([user2.userId]);
+        assert(res === undefined || res);
+    });
+
+    it('should get users presence', async () => {
+        const res = await client.getPresence([user2.userId]);
+        presence = res && res[0];
```

Comment:
In all these I don't think you need to check for a valid "res" as the above call would throw an exception is the getPresence call fails.

Reply:
done